tests/installation/confirm_installation_maintenance.pm

```diff
@@ -0,0 +1,28 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: This test module confirms the popup for installation and additionally, in case
+#          there is an update of a non-free license package, accepts the extra license popup
+#          that will appear before the confirm installation popup.
+
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+
+sub run {
+    my $install_popup = $testapi::distri->get_ok_popup_controller();
+    my $license_popup = $testapi::distri->get_license_popup();
+    # The first check is for the confirm installation popup, in order to save time as
+    # extra license popups will rarely appear.
+    unless ($install_popup->is_ok_popup_visible()) {
+        save_screenshot();
+        $license_popup->accept();
```

Comment:
I would have opt for a code without conditions here...one method that returns a handler to the page that detects, and then call `accept()` with it. ie: Javascript implements `race` to get the fastest thing to appear in the screen: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race

```
# tumbleweed.pm
sub get_any_popup {
...
}

# confirm_installation_maintenance.pm
my $any_popup = $testapi::distri->get_any_popup();
my $popup = $any_popup.race([$install_popup, $license_popup]);
$popup->accept();

# AnyPopup.pm
sub race {
 # calls to is_shown for each page passed, method try/catched or without die and win the best.
}
```

But it is just an idea in pseudocode, you don't need to go that path, we can accept your existing code.

Reply:
I have been checking this for a while, there is a race option in Mojo library : https://docs.mojolicious.org/Mojo/Promise which I could possibly use.
What I don't like is 
1) in case the java popup appears we need to accept both popups, in case it doesn't we accept only the second, so it's not either one or the other
2) I assume underneath the race functions there are quite a few if statements, so our code might look more elegant but on computing level doesn't seem like a big gain. 

Do we cover the expectations better? The current code in the PR expects to confirm installation, or if the license popup appears to fist accept it before the installation confirmation. In case of a race condition, we set the same expectations, no? So, I will not go forward with this now, but we can keep it in mind for the future.

Reply:
oh, cool, nice finding! mmm that opens a door, but would require to change lower layer of our own framework.
1 is the logic of the application, it is how it is, there is not accept all or something like that. About 2, complexity is always there, there is no magic way to get rid of it, code designs just move this complexity from one place to another. Relying in a solid tested "race" method is always more maintainable than creating our own hacky if/else condition each time (and sometimes in a different way each time).
No problem to go forward with your current implementation,
Could you please file a ticket with that idea for Mojo promises?